lib/kdtree.h

```diff
@@ -160,15 +160,46 @@ class Node {
 
 class KDTree {
 
-    using TriangleId = impl::TriangleId;
     using TriangleIds = impl::TriangleIds;
     using Node = impl::Node;
 
 public:
+
+    using TriangleId = impl::TriangleId;
+
+    class OptionalId {
```

Comment:
hm, boost::optional ist aber keine Monade ;). Ist wahrscheinlich auch etwas aufwändiger in C++.